test/unittest/unittest_nntrainer_appcontext.cpp

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file	unittest_app_context.h
+ * @date	9 November 2020
+ * @brief	This file contains app context related functions and classes that
+ * manages the global configuration of the current environment
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <fstream>
+#include <unistd.h>
+
+#include <app_context.h>
+
+class nntrainerAppContextDirectory : public ::testing::Test {
+
+protected:
+  void SetUp() override {
+    int status = mkdir("testdir", 0777);
+    ASSERT_EQ(status, 0);
+
+    std::ofstream file("testdir/testfile.txt");
+    ASSERT_EQ(file.fail(), false);
+
+    file << "testdata";
+    ASSERT_EQ(file.fail(), false);
+
+    file.close();
+
+    char buf[2048];
+    char *ret = getcwd(buf, 2048);
+    ASSERT_NE(ret, nullptr);
+    current_directory = std::string(buf);
+  }
+
+  void TearDown() override {
+    int status = remove("testdir/testfile.txt");
+    ASSERT_EQ(status, 0);
+
+    status = rmdir("testdir");
+    ASSERT_EQ(status, 0);
+  }
+
+  std::string current_directory;
+};
+
+TEST_F(nntrainerAppContextDirectory, readFromGetPath_p) {
+  nntrainer::AppContext ac = nntrainer::AppContext::Global();
+
+  std::string path = ac.getWorkingPath("testfile.txt");
+  EXPECT_EQ(path, "testfile.txt");
+
+  ac.setWorkingDirectory("testdir");
+
+  path = ac.getWorkingPath("testfile.txt");
+  EXPECT_EQ(path, current_directory + "/testdir/testfile.txt");
```

Comment:
as per this example, `setWorkingDirectory("testdir")` sets `${CWD}\testdir`.
Then, in that case, the default working directory must be `${CWD}`? but this is opposite to unittest on line 58.

When no working directory is set, CWD must be evaluated always with `getWorkingPath()` in case user changed it.

Reply:
I think user can expect it to be relative path if working path is not set. If we fixate to current `${CWD}`. changing `chdir()` would take no effect.(and also our itc, utc will break) It is intended behavior so it is tested. However, doc was wrong. My apologies.